### DISCUSSION TITLE
How are versions updated/managed in runtine/src/libraries?

### DISCUSSION THREAD

**Discussion by kikaragyozov** [dotnet/runtime#94461] [Q&A]
I'm trying to learn how exactly versions are being released for `Microsoft.DependencyInjection` and `Microsoft.DependencyInjection.Abstractions`. Both nuget packages, like any other nuget package, is in its own solution and folder, referencing other projects outside their own solution, if required.

Initially, I thought that the versions for both of these packages are always the same, i.e you bump up `Microsoft.DependencyInjection.Abstractions` if the need to bump up `Microsoft.DependencyInjection` arises, but then I noticed that `Microsoft.DependencyInjection` has a version `6.0.1`, which is not found in `Microsoft.DependencyInjection.Abstractions`. Okay, that means that's not what's happening here.

How exactly are versions managed, as I'd like to create repo for managing the versions of 2 such NuGet packages - one for abstractions, and another for abstractions + concrete implementations, all in 1 solution? (Or 2, if 1 will do more harm than good.)

**Comment by huoyaoyuan** [comment]
See also https://github.com/dotnet/runtime/blob/main/docs/project/library-servicing.md

**Reply by kikaragyozov** [reply]
How is the `ServicingVersion` tag used to generate a version?

**Reply by kikaragyozov** [reply]
I'd really appreciate a more in-depth explanation of how the whole orchestration with versioning works. The runtime is quite a large beast to tackle, so I'm aware that taking time to write an extensive quality post on how things work in tandem to accomplish versioning might not be desirable, but I'm sure enthusiasts like myself who stumble upon this would greatly appreciate the hard work.
